src/Random/Extra.elm

```diff
@@ -411,3 +429,86 @@ andThen6 constructor generatorA generatorB generatorC generatorD generatorE gene
                                     )
                         )
             )
+
+
+{-| Decide what steps to take next in your loop.
+
+If you are Done, you give the result of the whole loop. If you decide to Loop
+around again, you give a new state to work from. Maybe you need to add an item
+to a list? Or maybe you need to track some information about what you just saw?
+
+**Note**: It may be helpful to learn about [finite-state
+machines](https://en.wikipedia.org/wiki/Finite-state_machine) to get a broader
+intuition about using state. I.e. You may want to create a type that describes
+four possible states, and then use Loop to transition between them as you
+consume characters.
```

Comment:
Generators don't consume characters. I'm guessing this was copied from the `elm/parser` docs. We probably want to change the wording a bit here so it fits with `Random`.